zproject_java.gsl

```diff
@@ -814,7 +814,48 @@ implements AutoCloseable\
 }
 .endmacro
 
+.macro generate_jni_callback_c (argument)
+.    if type = "zactor_fn"
+void \
+.    else
+int \
+.    endif
+.    # TODO: verify that these don't duplicate
+$(type)_cb_wrapper(\
+.    # This method signature needs to be built on the model's knowledge of
+.    # the function callback type
+.    # Q: Where is that information available?
+.    if type = "zactor_fn"
+zsock_t* pipe, void* args\
+.    elsif type = "zloop_reader_fn"
+zloop_t *loop, zsock_t *reader, void* args\
+.    elsif type = "zloop_fn"
+zloop_t *loop, zmq_pollitem_t *item, void *arg\
+.    elsif type = "zloop_timer_fn"
+zloop_t *loop, int timer_id, void *arg\
+.    else
+garbage *unknown, void *arg\
+.    endif
+){
+    jni_cb_struct* wrapper = (jni_cb_struct)args;
+    JNIEnv* env = wrapper->env;
+    (*env)->Call\
+.    if type = "zactor_fn"
+Void\
+.    else
+Int\
+.    endif
+Method(env, wrapper->obj, wrapper->mid, wrapper->args);
```

Comment:
In order to make this happen with the current approach, we really need multiple ICallback variations. Each with an appropriate method signature.